lti_consumer/lti_1p1/oauth.py

```diff
@@ -147,7 +147,7 @@ def log_authorization_header(request, client_key, client_secret):
     oauth_body_hash = str(base64.b64encode(sha1.digest()))
     log.debug("[LTI] oauth_body_hash = %s", oauth_body_hash)
     client = oauth1.Client(client_key, client_secret)
-    params = client.get_oauth_params(request)
+    params = oauth1.rfc5849.signature.collect_parameters(headers=request.headers, exclude_oauth_signature=False)
```

Comment:
@Agrendalath Debugging the Oauth failure, I looked at the source for [Webob request](https://github.com/Pylons/webob/blob/259230aa2b8b9cf675c996e157c5cf021c256059/src/webob/request.py#L941) body attribute and the logic in the Oauthlib's `get_oauth_params` [function](https://github.com/oauthlib/oauthlib/blob/master/oauthlib/oauth1/rfc5849/__init__.py#L222). While Webob maintains the content of a request body as binary, oauthlib expects it to be a string.

Both this line here and the oauthlib's code are at least 7 years old and I am not sure how this mismatch didn't raise an issue before. Maybe because, this function was always called with `www-form-urlencoded` data?? 

Anyway, I found that there is alternate function which accpets binary body and can basically do the exact same function in [Line 92](https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p1/oauth.py#L92). I have reused it here as an replacement.

Since the `oauth_body_hash` is explicitly calculated in the above lines, I think we can consider this a safe replacement to get the other params.

Reply:
@tecoholic, good catch! It's working for me now.

However, this is not a part of the runtime usage refactor, so we should move this to a separate commit (or even PR).

Reply:
@Agrendalath I thought about this.
> 
> However, this is not a part of the runtime usage refactor, so we should move this to a separate commit (or even PR).

I think we can move this to a separate commit, but not a separate PR. I tried to see if we can test the rebinding using `www-form-urlencoded` requests and avoid body-hashing, but the rebinding is specifically for [JSON based score updates](https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_xblock.py#L1399) and it wouldn't be possible to test it without this tweak/fix.

Reply:
Please split this fix into a separate commit then. Leaving everything in a single commit will not give any idea of why we've changed it.

Reply:
@tecoholic, ^